src/static/cm_leaf_preparator.py

```diff
@@ -28,12 +29,16 @@ def _calculate_full_orig_cm(self) -> torch.Tensor:
         Returns: torch.Tensor: The full contact matrix.
         """
         contact_data = self.data.contact_data
```

Comment:
Create a `CMData` class, which loads the current model's contact matrix and calculates the full version. Then this logic will be there, not here.